admin/pre_reg/views.py

```diff
@@ -20,169 +17,172 @@
+        }
+
+
+class DraftDetailView(PreregAdmin, DetailView):
+    template_name = 'pre_reg/draft_detail.html'
+    context_object_name = 'draft'
+
+    def get(self, request, *args, **kwargs):
+        try:
+            return super(DraftDetailView, self).get(request, *args, **kwargs)
+        except AttributeError:
+            return handle_attribute_error(request, self.context_object_name,
+                                          self.kwargs.get('draft_pk'))
 
-@csrf_exempt
-@user_passes_test(is_in_prereg_group)
-def approve_draft(request, draft_pk):
-    """Approves current draft
-    :param request: mostly for user
-    :param draft_pk: Unique id for current draft
-    :return: DraftRegistrationApproval obj
-    """
-    draft = get_draft_or_error(draft_pk)
-
-    user = request.user.osf_user
-    draft.approve(user)
-    update_admin_log(
-        request.user.id, draft._id, 'Draft Registration',
-        'approved', action_flag=ACCEPT_PREREG
-    )
-    return redirect(reverse('pre_reg:prereg') + "?page={0}".format(request.POST.get('page', 1)), permanent=True)
-
-
-@csrf_exempt
-@user_passes_test(is_in_prereg_group)
-def reject_draft(request, draft_pk):
-    """Rejects current draft
-    :param request: mostly for user
-    :param draft_pk: Unique id for current draft
-    :return: DraftRegistrationApproval obj
-    """
-    draft = get_draft_or_error(draft_pk)
-
-    user = request.user.osf_user
-    draft.reject(user)
-    update_admin_log(
-        request.user.id, draft._id, 'Draft Registration',
-        'rejected', action_flag=REJECT_PREREG
-    )
-    return redirect(reverse('pre_reg:prereg') + "?page={0}".format(request.POST.get('page', 1)), permanent=True)
+    def get_object(self, queryset=None):
+        return serializers.serialize_draft_registration(
+            DraftRegistration.load(self.kwargs.get('draft_pk'))
+        )
 
 
-@csrf_exempt
-def update_draft(request, draft_pk):
-    """Updates current draft to save admin comments
+class DraftFormView(PreregAdmin, FormView):
+    template_name = 'pre_reg/draft_form.html'
+    form_class = DraftRegistrationForm
+    context_object_name = 'draft'
 
-    :param draft_pk: Unique id for current draft
-    :return: DraftRegistration obj
-    """
-    data = json.loads(request.body)
-    draft = get_draft_or_error(draft_pk)
+    def get(self, request, *args, **kwargs):
+        try:
+            return super(DraftFormView, self).get(request, *args, **kwargs)
+        except AttributeError:
+            return handle_attribute_error(request, self.context_object_name,
+                                          self.kwargs.get('draft_pk'))
 
-    if 'admin_settings' in data:
-        form = DraftRegistrationForm(data=data['admin_settings'])
-        if not form.is_valid():
-            return HttpResponseBadRequest("Invalid form data")
+    def post(self, request, *args, **kwargs):
+        try:
+            return super(DraftFormView, self).post(request, *args, **kwargs)
```

Comment:
From what I have seen this causes an nginx error somewhere in the 500s, I think because it is on the OSF side and not in Django. The error isn't recognized by Django. So I catch it and handle it myself.